run.py

```diff
@@ -832,6 +730,85 @@ def postprocess_all(settings):
         process_event_file(settings, year, iter)
 
 
+def to_singularity_volumes(volumes):
+    bindings = [f"{local_folder}:{binding['bind']}:{binding['mode']}" for local_folder, binding in volumes.items()]
+    result_str = ','.join(bindings)
+    return result_str
+
+
+def to_singularity_env(env):
+    bindings = [f"{env_var}={value}" for env_var, value in env.items()]
+    result_str = ','.join(bindings)
+    return '"' + result_str + '"'
+
+
+def run_container(client, settings: dict, image: str, volumes: dict, command: str,
+                  working_dir=None, environment=None):
+    """
+    Executes container using docker or singularity
+    :param client: the docker client. If it's provided then docker is used, otherwise singularity is used
+    :param settings: settings to get docker configuration
+    :param image: the image to run
+    :param volumes: a dictionary describing volume binding
+    :param command: the command to run
+    :param working_dir: the working directory inside the container. It's not necessary for docker because
+    docker file may have an instruction WORKDIR. In this case that directory is used. Singularity don't take this
+    instruction into account and the container working dir is the host working dir. Because of that most of the time
+     singularity requires working dir. One can get the work dir from a docker image by looking at the Dockerfile
+     (or image layers at the docker hub) and find the last WORKDIR instruction or by issuing a command:
+      docker run -it --entrypoint /bin/bash ghcr.io/lbnl-science-it/atlas:v1.0.7 -c "env | grep PWD"
+    :param environment: a dictionary that contains environment variables that needs to be set to the container
+    """
+    if client:
+        docker_stdout = settings.get('docker_stdout', False)
+        logger.info("Running docker container: %s, command: %s", image, command)
+        run_kwargs = {
+            'volumes': volumes,
+            'command': command,
+            'stdout': docker_stdout,
+            'stderr': True,
+            'detach': True
+        }
+        if working_dir:
+            run_kwargs['working_dir'] = working_dir
+        if environment:
+            run_kwargs['environment'] = environment
+        usim = client.containers.run(image, **run_kwargs)
```

Comment:
I think `usim` had been used as a name for the UrbanSim docker container, but this usage here is more generic. Maybe we can rename it to  something more general?

Reply:
Right I probably copied it from the Urbansim part